src/services/api.ts

```diff
@@ -1,9 +1,24 @@
-import axios, { AxiosInstance, InternalAxiosRequestConfig } from 'axios';
+import axios, { AxiosInstance, InternalAxiosRequestConfig, AxiosError, AxiosResponse } from 'axios';
+import { StatusCodes } from 'http-status-codes';
+import { toast } from 'react-toastify';
 import { getToken } from './token';
 
 const BACKEND_URL = 'https://15.design.htmlacademy.pro/six-cities';
 const REQUEST_TIMEOUT = 5000;
 
+type DetailMessageType = {
+  type: string;
+  message: string;
+}
+
+const StatusCodeMapping: Record<number, boolean> = {
+  [StatusCodes.BAD_REQUEST]: true,
+  [StatusCodes.UNAUTHORIZED]: true,
+  [StatusCodes.NOT_FOUND]: true
+};
+
+const shouldDisplayError = (response: AxiosResponse) => !!StatusCodeMapping[response.status];
```

Comment:
```suggestion
const errorStatusCodes = new Set<number>([
  StatusCodes.BAD_REQUEST,
  StatusCodes.UNAUTHORIZED,
  StatusCodes.NOT_FOUND
]);

const shouldDisplayError = (response: AxiosResponse): boolean => errorStatusCodes.has(response.status);
```

Использование Set делает проверку более быстрой и читаемой. В данном случае это подходит, так как мы просто проверяем наличие числа в наборе кодов.